packages/core/application/core-application-browser-internal/src/utils/get_location_observable.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Observable, Subject, startWith, shareReplay, distinctUntilChanged } from 'rxjs';
+import type { History } from 'history';
+
+// interface compatible for both window.location and history.location...
+export interface Location {
+  pathname: string;
+  hash: string;
+}
+
+export const getLocationObservable = (
+  initialLocation: Location,
+  history: History
+): Observable<string> => {
+  const subject = new Subject<string>();
+  history.listen((location) => {
+    subject.next(locationToUrl(location));
+  });
+  return subject.pipe(
+    startWith(locationToUrl(initialLocation)),
+    distinctUntilChanged(),
+    shareReplay(1)
+  );
```

Comment:
Emit with initial page's location + on each history change.  Using `distinctUntilChanged` to avoid emitting multiple time for the exact same url